psi4/src/psi4/libmints/molecule.h

```diff
@@ -364,13 +364,15 @@ class Molecule
     /// Computes center of mass of molecule (does not translate molecule)
     Vector3 center_of_mass() const;
     /// Computes nuclear repulsion energy
-    double nuclear_repulsion_energy() const;
+    /// The dipole field is a vector of length 3, containing the dipole field strength in the {x,y,z} directions.
+    double nuclear_repulsion_energy(std::vector<double> dipole_field) const;
```

Comment:
Maybe these should be `const std::array<double, 3>` then you can remove the length check in the function itself.

Reply:
Oh, that's a great idea.  I originally wanted `Vector3`, but didn't want to use a pointer, or include the header in `wavefunction.h`.  Using `std::array` gives the right behavior, with none of the drawbacks.